src/plugins/jni/CMakeLists.txt

```diff
@@ -10,7 +10,15 @@ if (DEPENDENCY_PHASE)
 
 	find_package (JNI)
 
-	if (JNI_FOUND)
+	# ~~~
+	# Disable plugin if full and shared build are enabled, since otherwise `kdb test` will fail printing the following error message:
+	#
+	#     kdb-full: error while loading shared libraries: libjvm.so: cannot open shared object file: No such file or directory
+	# .
+	# ~~~
+	if (BUILD_FULL AND BUILD_SHARED)
```

Comment:
> Is this maybe APPLE only?

Nope, in fact I only saw this error in the Travis [build job `🐧 FULL`](https://travis-ci.org/sanssecours/elektra/jobs/410850729#L3582).

> Does ONLY_SHARED maybe resolve the problem?

Oh, I did not now about the option. I will try to use that instead of the `if` statement.

Reply:
ONLY_SHARED was quite recently added by @e1528532 

For Haskell it helped and the problem might be very similar.